Dynatrace/dynatrace-operator
# Enable the usage of verbosity levels for logging

## Description

The main purpose of this PR is to straighten our logging and make the usage of verbosity levels (aka. `log.V(n)` possible. This also includes a bit of refactoring and code cleanup. This PR intentionally does not change existing log lines or introduce lot's of debug logging. That's for later. Also the actual log output should be kept as close as possible to what we have now.

This PR is split in two commits to make reviewing a bit easier. 
* 2952d2ad8ff45cb54c380004e256e27cc4527cd3 get's rid of a useless factory pattern. This one touches lot's of files. 
* The core refactoring is in 21e6c151d569468d089462fa67551a0f3b8c5e14.

## How can this be tested?

Deploy the operator and check if logs are still arriving and are readable (at least as readable as they used to be ;-)

## Checklist
